ksql-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java

```diff
@@ -169,6 +171,7 @@ public Statement build(final SingleStatementContext statement) {
     private final MetaStore metaStore;
 
     private int selectItemIndex = 0;
+    private boolean buildingPersistentQuery = false;
```

Comment:
unfortunate that we don't have a context to pass along instead of have it as field state :( would be a good place for this. anyway just a thought to see if you could do anything about it

Reply:
Yeah, but not much ROI for this right now...